rtgui/edit.cc

```diff
@@ -747,7 +747,294 @@ void Rectangle::drawToMOChannel(Cairo::RefPtr<Cairo::Context> &cr, unsigned shor
     }
 }
 
-EditSubscriber::EditSubscriber (EditType editType) : ID(EUID_None), editingType(editType), bufferType(BT_SINGLEPLANE_FLOAT), provider(NULL), dragging(false) {}
+void OPIcon::drivenPointToRectangle(rtengine::Coord &pos,
+                                    rtengine::Coord &topLeft, rtengine::Coord &bottomRight, int W, int H)
+{
+    switch (drivenPoint) {
+    case (DP_CENTERCENTER):
+        topLeft.x = pos.x - W / 2;
+        topLeft.y = pos.y - H / 2;
+        break;
+
+    case (DP_TOPLEFT):
+        topLeft.x = pos.x;
+        topLeft.y = pos.y;
+        break;
+
+    case (DP_TOPCENTER):
+        topLeft.x = pos.x - W / 2;
+        topLeft.y = pos.y;
+        break;
+
+    case (DP_TOPRIGHT):
+        topLeft.x = pos.x - W;
+        topLeft.y = pos.y;
+        break;
+
+    case (DP_CENTERRIGHT):
+        topLeft.x = pos.x - W;
+        topLeft.y = pos.y - H / 2;
+        break;
+
+    case (DP_BOTTOMRIGHT):
+        topLeft.x = pos.x - W;
+        topLeft.y = pos.y - H;
+        break;
+
+    case (DP_BOTTOMCENTER):
+        topLeft.x = pos.x - W / 2;
+        topLeft.y = pos.y - H;
+        break;
+
+    case (DP_BOTTOMLEFT):
+        topLeft.x = pos.x;
+        topLeft.y = pos.y - H;
+        break;
+
+    case (DP_CENTERLEFT):
+        topLeft.x = pos.x;
+        topLeft.y = pos.y - H / 2;
+        break;
+    }
+
+    bottomRight.x = topLeft.x + W - 1;
+    bottomRight.y = topLeft.y + H - 1;
+}
+
+OPIcon::OPIcon(Cairo::RefPtr<Cairo::ImageSurface> *normal,
```

Comment:
As those are never modified and `RefPtr` itself already indicates whether it contains a valid pointer (same code as now, just test it like a Boolean), I think these would usually be of type `const Cairo::RefPtr<Cairo::ImageSurface>&`. (Passing in an empty `RefPtr` using its default constructor via `{}` should not much less efficient but the interface becomes safer.)